pages/treeherder.py

```diff
@@ -31,12 +31,14 @@ class TreeherderPage(Base):
     _quick_filter_locator = (By.ID, 'quick-filter')
     _repos_menu_locator = (By.ID, 'repoLabel')
     _result_sets_locator = (By.CSS_SELECTOR, '.result-set:not(.row)')
+    _resultset_visible_locator = (By.CSS_SELECTOR, 'span.hidden')
     _unchecked_repos_links_locator = (By.CSS_SELECTOR, '#repoLabel + .dropdown-menu .dropdown-checkbox:not([checked]) + .dropdown-link')
     _unclassified_failure_count_locator = (By.ID, 'unclassified-failure-count')
     _unclassified_failure_filter_locator = (By.CSS_SELECTOR, '.btn-unclassified-failures')
 
     def wait_for_page_to_load(self):
-        self.wait.until(lambda s: self.unclassified_failure_count > 0)
+        el = self.find_element(*self._resultset_visible_locator)
+        self.wait.until(lambda s: 'hidden' in el.get_attribute('class'))
```

Comment:
I tried every variation I could think of for this with span.hidden and the .ready state, in addition waiting for the class to include ready. I also tried with ng-class. All of the tests continued to close the browser before jobs had loaded. Instead of waiting for the unclassified failure count in the header, I changed it to check that a result set has loaded in the page which more closely matches what #th-global-content span.hidden.ready is supposed to verify.